scripts/speaker_tasks/eval_diar_with_asr.py

```diff
@@ -0,0 +1,269 @@
+ --root_path="/path/to/output/directory"
+```
+
+Only use RTTMs to calculate DER
+```
+python eval_diar_with_asr.py \
+ --hyp_rttm_list="/path/to/hypothesis_rttm_filepaths.list" \
+ --ref_rttm_list="/path/to/reference_rttm_filepaths.list" \
+ --root_path="/path/to/output/directory"
+```
+
+"""
+
+
+def get_pyannote_objs_from_rttms(rttm_file_path_list):
+    """Generate PyAnnote objects from RTTM file list
+    """
+    pyannote_obj_list = []
+    for rttm_file in rttm_file_path_list:
+        rttm_file = rttm_file.strip()
+        if rttm_file is not None and os.path.exists(rttm_file):
+            uniq_id = get_uniqname_from_filepath(rttm_file)
+            ref_labels = rttm_to_labels(rttm_file)
+            reference = labels_to_pyannote_object(ref_labels, uniq_name=uniq_id)
+            pyannote_obj_list.append([uniq_id, reference])
+    return pyannote_obj_list
+
+
+def make_meta_dict(hyp_rttm_list, ref_rttm_list):
+    """Create a temporary `audio_rttm_map_dict` for evaluation
+    """
+    meta_dict = {}
+    for k, rttm_file in enumerate(ref_rttm_list):
+        uniq_id = get_uniqname_from_filepath(rttm_file)
+        meta_dict[uniq_id] = {"rttm_filepath": rttm_file.strip()}
+        if hyp_rttm_list is not None:
+            hyp_rttm_file = hyp_rttm_list[k]
+            meta_dict[uniq_id].update({"hyp_rttm_filepath": hyp_rttm_file.strip()})
+    return meta_dict
+
+
+def make_trans_info_dict(hyp_json_list_path):
+    """Create `trans_info_dict` from the `.json` files
+    """
+    trans_info_dict = {}
+    for json_file in hyp_json_list_path:
+        json_file = json_file.strip()
+        with open(json_file) as jsf:
+            json_data = json.load(jsf)
+        uniq_id = get_uniqname_from_filepath(json_file)
+        trans_info_dict[uniq_id] = json_data
+    return trans_info_dict
+
+
+def read_file_path(list_path):
+    """Read file path and strip to remove line change symbol
+    """
+    return sorted([x.strip() for x in read_file(list_path)])
+
+
+def evaluate_der(audio_rttm_map_dict, all_reference, all_hypothesis, diar_eval_mode):
+    """Evaluate with a selected diarization evaluation scheme
+    """
+    eval_settings = []
+    if diar_eval_mode == "full":
+        eval_settings = [(0.0, False)]
+    elif diar_eval_mode == "fair":
+        eval_settings = [(0.25, False)]
+    elif diar_eval_mode == "forgiving":
+        eval_settings = [(0.25, True)]
+    elif diar_eval_mode == "all":
+        eval_settings = [(0.0, False), (0.25, False), (0.25, True)]
+    else:
+        raise ValueError("`diar_eval_mode` variable contains an unsupported value")
+
+    for collar, ignore_overlap in eval_settings:
+        diar_score = score_labels(
+            AUDIO_RTTM_MAP=audio_rttm_map_dict,
+            all_reference=all_reference,
+            all_hypothesis=all_hypothesis,
+            collar=collar,
+            ignore_overlap=ignore_overlap,
+        )
+    return diar_score
```

Comment:
shall we keep this part of speaker_utils?

Reply:
moved to der.py